### ISSUE TITLE
Subs Only - Request, not Issue

### ISSUE THREAD

**Issue by streetsy** [Ascoware/get-iplayer-automator#59]
Would it be possible to have GiA only download Subtitles, and not the Video file? Perhaps via a checkbox?
Alternatively, could it download and convert subs first, and then the video (so I could then cancel and just get the subs)?
This is especially relevant on ITV, where, if using other ITV players, they are not currently capable of getting the subs, so I would like to download them via GiA, without having to download the whole episode.

**Comment by streetsy** [comment]
Ok, having received no reply (and I know that maintaining this involves a lot of work, and I'm very grateful to skovatch and other contributors), I decided to have a go at sorting my issue myself.
I am NOT good enough at coding, so this is what I have done:

I select the ITV program I want subs for in GiA. I then use the "Show Log" button, so the log window is open. Then I start the download, and immediately stop it.
I go to the log window, and, in there, I find the subtitles url. I copy all of it, and paste it into a browser.
I use "save as" in the browser to save the xml file to a Subtitles folder on my Mac.
Separately, I used Automator to create a Workflow which I saved as a Service (SubstoSrt). That workflow uses the Run Shell Script action on the ttml2srt.py file that is included in GiA, and of which I saved a copy in my Subtitles folder.
Once I visited the xml page in the browser and saved it, I go to the file on my mac, right-click it, and select the service I created. It immediately converts it to srt for me.
I then share that folder using samba (SMBUp) with my mediacenter, so I can "Browse for Subtitle" and watch my ITV program on my telly.

I know this is a convoluted way of doing things, and that it would all get done within GiA anyway. My issue with GiA is that it doesn't get, convert, and save, the subtitles file until AFTER it downloads the programme. However, I don't WANT to download the programme, as I view it via an ITV streaming plugin in my Mediacenter. The streaming plugin, however, is unable to get the subtitles since a while back, as ITV changed the way the subtitles are obtained on their website.

Apologies for the long post, just thought it may help others if they are also a bit aurally-challenged like me!

**Comment by hugofalkman** [comment]
@streetsy the command line tool get_iplayer (that GiA uses) has a subtitles only option. Sounds like that would be easier for you to use.

**Comment by streetsy** [comment]
Thanks @hugofalkman, you're correct with your edit, though, only works for BBC and the BBC plugins I use are OK with getting subs, it's only ITV I'm stuck with.

**Comment by skovatch** [comment]
Sorry, I've been away recovering from eye surgery this week. This is a completely valid request, and since ITV downloads are done entirely within GiA I'll take a look. You've got the right logic; I would just need to add a preference.

**Comment by streetsy** [comment]
Skovatch,
Thank you so much...the priority should be to recover well from your surgery, though!
Take care, and thanks again for everything you do with this project.

**Comment by Lordsnooty05** [comment]
Yes streetsy I totally agree Skovatch priority must be his health over everything else....get well soon. I’m using a work around by just downloading one ITV programme at a time because it seems to get stuck after one download.......yes I know it is totally weird but that’s what it does. Luckily there isn’t much quality programs on ITV.......sorry , no there isn’t ....lol

**Comment by demosthien** [comment]
@Skovatch
Could I add a possible adjustment to streetsy's sub dl request?

Perhaps making a case-by-case option would be worthwhile?

When item is in PVR having checkboxes for video and sub which (default as per preferences) can be changed for scheduled dl.

And when the item is manually added to the dl queue having those checkboxes for unscheduled dl.

Example of this for me is that I only want subs for Inspector Montalbano (native audio is Italian) but right now I have to dl subs for everything.

**Comment by hugofalkman** [comment]
@demosthien Think BBC for programmes in other languages than English, such as Inspector Montalbano, put the subtitles directly into the video track. There is still a separate subtitle track to download for the hearing impaired, containing additional subtitles such as "music in the background", "gun shots" etc.
